### PR TITLE
add new colors. shift some classes to use the new colors. update the …

### PR DESCRIPTION
**Pull request by kentmclean**
…visual of the colors page

### CODE REVIEW

File: src/app.jsx [hunk 1]
```diff
@@ -1,17 +1,17 @@
-var React = require('react');
-var ReactDOM = require('react-dom');
-var { Router, Route, Redirect, Link } = require('react-router');
-var components = require('./components.js');
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { HashRouter, Route, Redirect, NavLink } from 'react-router-dom';
```
Comment (kentmclean) [inline src/app.jsx]: Upgraded router. Most all changes within this file are directly a result of that.

File: src/app.jsx [hunk 2]
```diff
@@ -54,7 +54,7 @@ var Sidebar = function ({ route }) {
 	);
 }
 
-var TopNav = ({ route }) => {
+const TopNav = ({ match }) => {
 	return (
 		<div>
 			<div className="cps-topnav" style={{ position: 'fixed', width: '100%', height: '38px', backgroundColor: '#00bf4b', top: 0, zIndex: 100 }}>
```
Comment (kentmclean) [inline src/app.jsx]: I'll probably remove this whole topnav in a future iteration.

File: src/components/typography/typography.jsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ module.exports = class extends React.Component{
 						<div className="cps-col-xs-2">
 							<span className="cps-pull-right">.cps-wt-light</span>
 						</div>
-						<div className="cps-col-xs-8 cps-text-muted-dark">An economy based on endless growth is unsustainable</div>
```
Comment (kentmclean) [inline src/components/typography/typography.jsx]: Just removed some unused classes

File: src/components/typography/typography.less [hunk 4]
```diff
@@ -59,36 +59,3 @@ h1,h2,h3,h4,h5,h6 {
 strong {
 	.cps-wt-semibold;
 }
-
-.gen-color-class(warning, @warning);
```
Comment (kentmclean) [inline src/components/typography/typography.less]: moved this block into the `colors.less` file

File: src/components/variables/colors.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+const colorCategories = {
+	brand: 'Brand',
+	accents: 'Accents',
+	gray: 'Gray',
+	base: 'Base',
+};
+
+const colors = {
+	[colorCategories.brand]: {
```
Comment (kentmclean) [inline src/components/variables/colors.js]: desire is to make this or a similar definition of the styles drive both the documentation side as is done here *as well as* the creation of the classes themselves. Maybe talk with the group around JS-to-CSS options there

File: src/components/variables/colors.less [hunk 6]
```diff
@@ -1,6 +1,61 @@
...
+
+.gen-color-class(gray-1, @gray-1);
+.gen-color-class(gray-2, @gray-2);
+.gen-color-class(gray-3, @gray-3);
+.gen-color-class(gray-4, @gray-4);
+.gen-color-class(gray-5, @gray-5);
+.gen-color-class(gray-6, @gray-6);
+
+
+/************ Legacy definitions ************/
```
Comment (kentmclean) [inline src/components/variables/colors.less]: Keeping existing classes around

File: src/custom-elements/cps-loader/cps-loader.helper.js [hunk 7]
```diff
@@ -1,33 +1,31 @@
-import {h} from 'preact';
```
Comment (kentmclean) [inline src/custom-elements/cps-loader/cps-loader.helper.js]: spaces converted to tabs

File: package.json [hunk 8]
```diff
@@ -33,41 +33,42 @@
...
+    "karma-webpack": "^2.0.4",
     "less": "2.7.2",
     "less-loader": "4.0.5",
     "lodash": "^4.17.4",
     "postcss-loader": "^2.0.6",
-    "react": "^0.14.0",
-    "react-dom": "^0.14.0",
-    "react-highlight": "^0.9.0",
-    "react-router": "1.0.3",
+    "react": "^16.0.0",
```
Comment (kentmclean) [inline package.json]: updated to 16!

Comment (neiloler) [reply package.json]: Awwww yussss.

File: src/components/variables/colors.js (hunk 5 above)
Comment (neiloler) [reply src/components/variables/colors.js]: This is an interesting idea. Not entirely sure I can see all the implications, but I do like the coding/dynamic nature of this.

File: src/components/core.less [hunk 9]
```diff
@@ -21,11 +21,12 @@ html {
 body {
 	margin: 0;
 	padding-top: 0px;
-	background-color: #f4f5f8;
+	padding-bottom: 30px;
```
Comment (joeldenning) [inline src/components/core.less]: Why are we putting this back in? This was removed with https://github.com/CanopyTax/canopy-styleguide/pull/96/files

Maybe a bad merge?

**Pull request merged**